pipelines/common/src/main/java/com/google/fhir/analytics/view/ViewManager.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020-2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.fhir.analytics.view;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Multimap;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+
+/**
+ * This class is responsible for reading ViewDefinition files and producing the relevant row for
+ * each input resource based on the views that apply to that resource.
+ */
+public class ViewManager {
+
+  private static final String JSON_EXT = ".json";
+
+  private final Multimap<String, ViewDefinition> viewMap = HashMultimap.create();
+
+  // This should only be instantiated with factory methods.
+  private ViewManager() {}
+
+  // Note we rely on JAVA NIO API for file access; this may require additional dependencies to work
+  // on distributed file systems, e.g., https://github.com/googleapis/java-storage-nio
```

Comment:
How about using Beam file system APIs - `org.apache.beam.sdk.io`?

Reply:
As discussed off-PR, this will add a Beam dependency for this `view` library which is not desirable. To some extent I have changed my mind re. depending on Beam file-system (in general); especially when it adds a new dependency.

Reply:
Yes, it makes sense to not add a dependency on the `common` module. Additionally, if we are planning to separate out the `View` package as well, then this helps.